src/vivarium_gates_nutrition_optimization_child/components/wasting.py

```diff
@@ -0,0 +1,857 @@
+
+
+class WastingTreatment(Risk):
+    def __init__(self, treatment_type: str):
+        super().__init__(treatment_type)
+
+        self.previous_wasting_column = f"previous_{data_keys.WASTING.name}"
+        self.wasting_column = data_keys.WASTING.name
+
+        self.treated_state = self._get_treated_state()
+
+    ##########################
+    # Initialization methods #
+    ##########################
+
+    def _get_treated_state(self) -> str:
+        return self.risk.name.split("_treatment")[0]
+
+    #################
+    # Setup methods #
+    #################
+
+    def setup(self, builder: Builder):
+        super().setup(builder)
+        self._register_on_time_step_prepare_listener(builder)
+
+    def _get_population_view(self, builder: Builder) -> PopulationView:
+        return builder.population.get_view(
+            [self.propensity_column_name, self.previous_wasting_column, self.wasting_column]
+        )
+
+    def _register_on_time_step_prepare_listener(self, builder: Builder) -> None:
+        builder.event.register_listener("time_step__cleanup", self.on_time_step_cleanup)
+
+    ########################
+    # Event-driven methods #
+    ########################
+
+    def on_time_step_cleanup(self, event: Event):
+        pop = self.population_view.get(event.index)
+        propensity = pop[self.propensity_column_name]
+        remitted_mask = (pop[self.previous_wasting_column] == self.treated_state) & pop[
+            self.wasting_column
+        ] != self.treated_state
+        propensity.loc[remitted_mask] = self.randomness.get_draw(remitted_mask.index)
+        self.population_view.update(propensity)
+
+
+class DynamicChildWastingModel(DiseaseModel):
+    def setup(self, builder):
+        """Perform this component's setup."""
+        super(DiseaseModel, self).setup(builder)
+
+        self.configuration_age_start = builder.configuration.population.age_start
+        self.configuration_age_end = builder.configuration.population.age_end
+
+        cause_specific_mortality_rate = self.load_cause_specific_mortality_rate_data(builder)
+        self.cause_specific_mortality_rate = builder.lookup.build_table(
+            cause_specific_mortality_rate,
+            key_columns=["sex"],
+            parameter_columns=["age", "year"],
+        )
+        builder.value.register_value_modifier(
+            "cause_specific_mortality_rate",
+            self.adjust_cause_specific_mortality_rate,
+            requires_columns=["age", "sex"],
+        )
+
+        self.population_view = builder.population.get_view(
+            ["age", "sex", self.state_column, "static_child_wasting_propensity"]
+        )
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants,
+            creates_columns=[self.state_column],
+            requires_columns=["age", "sex", "static_child_wasting_propensity"],
+            requires_streams=[f"{self.state_column}_initial_states"],
```

Comment:
We don't need this stream? Everything should be deterministic here.

Reply:
removed